projects/table-builder/src/lib/classes/table-store.ts

```diff
@@ -0,0 +1,192 @@
+import { FieldType, MetaData } from '../interfaces/report-def';
+import { v4 as uuid } from 'uuid';
+import { Observable } from 'rxjs';
+import { defaultTableState, TableState } from './TableState';
+import { Injectable, Inject } from '@angular/core';
+import { TableBuilderConfig, TableBuilderConfigToken } from './TableBuilderConfig';
+import { FilterInfo } from './filter-info';
+import { selectStorageStateItem, StateStorage } from '../ngrx/reducer';
+import { Sort, SortDirection }  from '@angular/material/sort' ;
+import { ComponentStore } from '@ngrx/component-store'  ;
+import update from 'immutability-helper';
+import { Dictionary } from '../interfaces/dictionary';
+import { select, Store } from '@ngrx/store';
+import { first, mergeMap, tap } from 'rxjs/operators';
+import { loadState, saveState } from '../ngrx/actions';
+
+
+@Injectable()
+export class TableStore extends ComponentStore<TableState> {
+
+  constructor(@Inject(TableBuilderConfigToken) private config: TableBuilderConfig,
+  private store: Store<{storageState: StateStorage}>) {
+    super( { ...defaultTableState, ...config.defaultTableState});
+  }
+  readonly filters$ = this.select(state => state.filters );
+
+  getFilter$(filterId) : Observable<FilterInfo | undefined> {
+    return this.select( this.filters$, filters => filters[filterId]);
```

Comment:
What is the value of `filters` in `defaultTableState`?  Can it ever be `undefined`, and throw an exception when you do `filters[filterId]`?

Reply:
filters is an object or Dictionary<FilterInfo>
In theory you can probably set it to null, but I don't think we need to worry about that